src/AttendanceHandler.cpp

```diff
@@ -0,0 +1,69 @@
+
+#include <algorithm>
+#include "AttendanceHandler.hpp"
+
+auto AttendanceHandler::matchByEmail(const User& userProfile)
+{
+    return [&userProfile](const auto& report)
+    {
+        return !report.second.getEmail().compare(userProfile.getEmail());
+    };
+}
+
+void AttendanceHandler::reportAbsence(const User& userProfile,
+                                      const Lesson& Lesson,
+                                      const std::string& comment)
+{
+    report(userProfile, Lesson, "Absence", comment);
+}
+
+void AttendanceHandler::reportMakeUp(const User& userProfile,
+                                     const Lesson& Lesson,
+                                     const std::string& absenceDate,
+                                     const std::string& comment)
+{
+    report(userProfile, Lesson, "MakeUp for: " + absenceDate, comment);
+}
+
+AttendanceHandler::AttendanceList AttendanceHandler::viewReports(const User * const userProfile)
+{
+    if(userProfile == nullptr) {
```

Comment:
if (not userProfile)

Reply:
I prefer to use "if (not userProfile)" only for variables of type bool
and for any other comparison "if(userProfile == nullptr) 

for me, it's more descriptive.

but as requested changed